core/src/main/java/org/springframework/security/authorization/AuthorizationManagers.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authorization;
+
+/**
+ * A factory class to create an {@link AuthorizationManager} instances.
+ *
+ * @author Evgeniy Cheban
+ * @since 5.8
+ */
+public final class AuthorizationManagers {
+
+	/**
+	 * Creates an {@link AuthorizationManager} that grants access if at least one
+	 * {@link AuthorizationManager} granted or abstained, if <code>managers</code> are
+	 * empty then {@link AuthorizationDecision#DENIED} is returned.
+	 * @param <T> the type of object that is being authorized
+	 * @param managers the {@link AuthorizationManager}s to use
+	 * @return the {@link AuthorizationManager} to use
+	 */
+	@SafeVarargs
+	public static <T> AuthorizationManager<T> anyOf(AuthorizationManager<T>... managers) {
+		return (authentication, object) -> {
+			for (AuthorizationManager<T> manager : managers) {
+				AuthorizationDecision decision = manager.check(authentication, object);
+				if (decision == null || decision.isGranted()) {
+					return decision;
+				}
+			}
+			return AuthorizationDecision.DENIED;
```

Comment:
Instead of returning `DENIED` here, I think we should return a composite authorization decision. Several authorization managers return extensions of `AuthorizationDecision` that have additional decision context, and we don't want to lose that.

You might consider adding `CompositeAuthorizationDecision` and returning an instance that contains all of the denial decisions.